packages/plugin/tests/no-duplicate-fields.spec.ts

```diff
@@ -48,7 +48,58 @@ ruleTester.runGraphQLTests('no-duplicate-fields', rule, {
           }
         }
       `,
-      errors: [{ message: 'Field `email` defined multiple times.' }],
+      errors: [{ message: 'Field `email` already defined.' }],
+    },
+    {
+      code: /* GraphQL */ `
+        {
+          users {
+            id
+            ...UserFullFields # email #email
+            name #3
+            ...UserFields # email firstName
+            ... on User {
+              id #6
+              name #7
+              email #8
+            }
```

Comment:
@B2o5T imagine for some nested UI components it is likely you are using `UserFullFields` in one place and `UserFields` in some other place and it can actually be a valid use case but this rule would mark that as error